addons/sale_crm/wizard/crm_opportunity_to_quotation.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class Opportunity2Quotation(models.TransientModel):
+
+    _name = 'crm.quotation.partner'
+    _description = 'Create new or use existing Customer on new Quotation'
+    _inherit = 'crm.partner.binding'
+
+    action = fields.Selection([
+        ('create', 'Create a new customer'),
+        ('exist', 'Use an existing customer')
+    ], 'Quotation Customer', required=True)
+
+    @api.multi
+    def action_apply(self):
+        """ Convert lead to opportunity or merge lead and opportunity and open
+            the freshly created opportunity view.
+        """
+        self.ensure_one()
+        lead = self.env['crm.lead'].browse(self._context.get('active_id'))
```

Comment:
wouldn't it be cleaner to create a `lead_id` field on this wizard ? I don't like having to include the active_id in the business method.